packages/medusa/src/models/batch-job.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  BeforeInsert,
+  Column,
+  CreateDateColumn,
+  DeleteDateColumn,
+  Entity,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from "typeorm"
+import { ulid } from "ulid"
+import { BatchJobStatus, BatchJobType } from "../types/batch-job"
+import { DbAwareColumn, resolveDbType } from "../utils/db-aware-column"
+
+@Entity()
+export class BatchJob {
+  @PrimaryColumn()
+  id: string
+
+  @DbAwareColumn({ type: "enum", enum: BatchJobType })
+  type: BatchJobType
```

Comment:
todo: this field should be open for different types than the ones defined in `BatchJobType` so just make it `type: "text"`